app/javascript/react/components/Button/Button.tsx

```diff
@@ -0,0 +1,13 @@
+import * as S from "./Button.style";
+import React from "react";
+
+interface ButtonProps {
+  onClick: () => void;
+  children: string;
```

Comment:
Can be a text node or other element like an icon, etc. This should work:
```suggestion
  children: string | JSX.Element | (JSX.Element | string)[];
```